tests/testthat/test_pipeline-helpers.R

```diff
@@ -67,6 +67,28 @@ test_that("protect and unprotect integer cols work", {
   expect_equal(names(d3), c("iso", "2050"))
 })
 
+test_that("left_join_error_no_match works", {
+
+  # Basic join works
+  x <- tibble(year = 2000:2002)
+  y <- tibble(year = 2000:2002, value = 1:3)
+  expect_silent(left_join_error_no_match(x, y, by = "year"))
+  expect_identical(left_join_error_no_match(x, y, by = "year"),
+                   left_join(x, y, by = "year"))
+
+  # Catches duplicate matches
+  y <- tibble(year = c(2000, 2001, 2001, 2002), value = 1:4)
+  expect_error(left_join_error_no_match(x, y, by = "year"))
+
+  # Catch no match (NAs)
```

Comment:
The only other test I can maybe think of is an explicit one for the number of rows changing, similar to this test. Since those are the two if conditions that _should_ throw errors based on the function.